components/create_post/create_post.jsx

```diff
@@ -536,21 +544,48 @@ class CreatePost extends React.PureComponent {
         const {
             currentChannel: updateChannel,
             userIsOutOfOffice,
+            allowReferencedGroups,
+            channelMemberCountsByGroup,
+            currentChannelMembersCount
         } = this.props;
 
-        const currentMembersCount = this.props.currentChannelMembersCount;
         const notificationsToChannel = this.props.enableConfirmNotificationsToChannel && this.props.useChannelMentions;
+        let memberCount = 0;
+        let timezoneCount = 0;
+        let groups = [];
+        if (this.props.enableConfirmNotificationsToChannel && !containsAtChannel(this.state.message)) {
+            // Groups mentioned in users text
+            groups = groupsMentionedInText(this.state.message, allowReferencedGroups);
+            if (groups.length > 0) {
+                const groupsWith5MembersInChannel = groups.map((group) => channelMemberCountsByGroup[group.id]).filter((group) => group);
+                groups = groups.map((group) => `@${group.name}`);
+                groupsWith5MembersInChannel.forEach((group) => {
+                    if (group.channel_member_count > Constants.NOTIFY_ALL_MEMBERS && group.channel_member_count > memberCount) {
+                        memberCount = group.channel_member_count;
+                        timezoneCount = group.channel_member_timezones_count;
+                    }
+                });
+            }
+        }
+
         if (notificationsToChannel &&
-            currentMembersCount > Constants.NOTIFY_ALL_MEMBERS &&
+            currentChannelMembersCount > Constants.NOTIFY_ALL_MEMBERS &&
             containsAtChannel(this.state.message)) {
+            memberCount = currentChannelMembersCount - 1;
+            groups = ['@all', '@channel'];
             if (this.props.isTimezoneEnabled) {
                 const {data} = await this.props.actions.getChannelTimezones(this.props.currentChannel.id);
-                if (data) {
-                    this.setState({channelTimezoneCount: data.length});
-                } else {
-                    this.setState({channelTimezoneCount: 0});
-                }
+                timezoneCount = data ? data.length : 0;
             }
+        }
+
+        if (memberCount > 0) {
```

Comment:
shouldnt this be `memberCount > 5`?

Reply:
oh nvm I understand - this value is only from 0 to something else if that value is >5